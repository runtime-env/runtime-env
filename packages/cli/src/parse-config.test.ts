import parseConfig from "../src/parse-config";

const minimalValidConfig = {
  globalVariableName: "_",
  envSchemaFilePath: "_",
};

describe("globalVariableName", () => {
  test("valid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        globalVariableName: "fooBar",
      }),
    ).not.toThrow();
  });

  test("invalid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        globalVariableName: "foo-bar",
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});

describe("genJs.*.mode", () => {
  test("valid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "bar",
          },
        ],
      }),
    ).not.toThrow();
  });

  test("invalid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
        ],
      }),
    ).toThrowErrorMatchingSnapshot();
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "bar",
          },
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
        ],
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});

describe("genJs.*.envFilePath", () => {
  test("valid 1", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: null,
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
        ],
      }),
    ).not.toThrow();
  });

  test("valid 2", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: "_",
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
        ],
      }),
    ).not.toThrow();
  });

  test("valid 3", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: ["_", "__"],
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
        ],
      }),
    ).not.toThrow();
  });

  test("invalid 1", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            envFilePath: [],
            userEnvironment: false,
            outputFilePath: "_",
            mode: "foo",
          },
        ],
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});
