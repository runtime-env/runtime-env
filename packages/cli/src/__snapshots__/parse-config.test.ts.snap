// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genJs.*.envFilePath invalid 1 1`] = `
"[
  {
    "code": "too_small",
    "minimum": 1,
    "type": "array",
    "inclusive": true,
    "exact": false,
    "message": "Array must contain at least 1 element(s)",
    "path": [
      "genJs",
      0,
      "envFilePath"
    ]
  }
]"
`;

exports[`genJs.*.mode invalid 1`] = `
"[
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: No duplicate modes allowed",
    "path": [
      "genJs",
      0,
      "mode"
    ]
  },
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: No duplicate modes allowed",
    "path": [
      "genJs",
      1,
      "mode"
    ]
  }
]"
`;

exports[`genJs.*.mode invalid 2`] = `
"[
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: No duplicate modes allowed",
    "path": [
      "genJs",
      0,
      "mode"
    ]
  },
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: No duplicate modes allowed",
    "path": [
      "genJs",
      2,
      "mode"
    ]
  }
]"
`;

exports[`globalVariableName invalid 1`] = `
"[
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: Invalid variable name",
    "path": [
      "globalVariableName"
    ]
  }
]"
`;
